test/combinations.jl

```diff
@@ -117,17 +117,20 @@ This file includes tests for:
         end # for
 
         iter_diff_methods = (
-            (supports.autoenzyme, AutoEnzyme()),
+            (true, FiniteDifference()),
+            (supports.autoenzyme, AutoEnzyme())
         )
 
-        for (supported, diff_method) in iter_diff_methods
+        for (supported, method) in iter_diff_methods
+            f = testable.integrand
+            geometry = testable.geometry
+            sol = testable.solution
+
             if supported
-                @test integral(
-                    testable.integrand, testable.geometry; diff_method = diff_method)≈testable.solution rtol=rtol
+                @test integral(f, geometry; diff_method = method)≈sol rtol=rtol
                 @test MeshIntegrals.supports_autoenzyme(testable.geometry) == true
```

Comment:
This should only be tested with the `AutoEnzyme()` backend shouldn't it?

Reply:
This `@testsnippet` imports Enzyme, so the previous `@test integral` that did not specify a `diff_method` should fall back to the default, which will use `AutoEnzyme()` (except for the few blacklisted cases).

This specific grouping of tests is intended to explicitly iterate through all `diff_method` choices, so I added `FiniteDifference()` to the iters set.

Reply:
I guess we're actually also missing user-friendly errors for situations where `AutoEnzyme` is specified/defaulted but isn't loaded and supported.

Reply:
What I mean is that this test will fail (and actually fails in CI) for geometries that do not support Enzyme (e.g., `BezierCurve`), when `method` is `FiniteDifference`.

Reply:
I just pushed a commit with changes to the `combinations.jl` tests that I'm hoping will fix this. Instead of relying on the package to tell us the correct support state that we will verify, which could lead to false negatives, I just added a keyword-argument flag triggered in the relevant `@testitem`s.

Reply:
I don't think this will fix it because above you explicitly set `supported` to always `true` in the `FiniteDifference()` case. The `supports` SupportStatus is not used for the test in this line.